src/BaseControllerV2.ts

```diff
@@ -9,6 +9,15 @@ import type { Draft } from 'immer';
  */
 export type Listener<T> = (state: T) => void;
 
+export type Anonymizer<T> = (value: T) => T;
+
+export type Schema<T> = {
+  [P in keyof T]: {
+    persist: boolean;
+    anonymous: boolean | Anonymizer<T[P]>;
+  };
+};
+
```

Comment:
1. This type (or interface, see below) should have a docstring explaining what the `anonymous` and `persist` values mean. From the usage, I take it that `anonymous: true` means that the corresponding state key contains no PII/sensitive data, but that wasn't immediately clear to me. 
2. Should this be an interface? Due to advances since TypeScript 2.7, I don't think we'll miss any relevant functionality, so this is mostly a question of style. Discussion: https://stackoverflow.com/questions/37233735/typescript-interfaces-vs-types

Reply:
I think a type makes more sense here? I don't know how to create an interface with properties "[P in keyof T]", so I'm not sure this would be supported. Plus, this doesn't seem very... interface-y to me? Interfaces are for duck typing. Here we're expecting a _very specific_ shape - _exactly_ the same properties as the state itself, so that the type prevents us from forgetting to list a state property.

Definitely needs a doc string though! I'll work on that.

Reply:
Oh, though now that you mention it, `Anonymizer` and `Listener` should probably be interfaces :thinking: I'll try that.

Reply:
Doc comments added in 9a67b9e22613c27062475ca1400df0b4279a08ee